src/client/ImageStore.js

```diff
@@ -54,6 +62,33 @@ export class Image {
     this.marked = false;
   }
 
+  // Updates the whole .data of this image
+  @action updateData() {
+    const imageId = this.data._id;
+    return axios.get(`/v1/image/${imageId}/details`).then(response => {
+      this.data = response.data;
+    })
+  }
+
+
+  @action changeAuthor(newCid) {
+    const imageId = this.data._id;
+    return axios.post(`/v1/image/${imageId}/author`, {newCid: newCid}).then(() => {
+        // Fetch the newly written parameters
+        return axios.get(`/v1/image/${imageId}/author`).then(response => {
+            console.log("New author: " + response.data);
+            this.data.author = response.data;
```

Comment:
Interesting, this works without having to `bind(this)` in order to capture `this` in the lambda?